### PR TITLE
error_injection: Re-use enter() code in inject() overloads

### PR DESCRIPTION
**Pull request by xemul**
Most of inject() overloads check if the injection is enabled, then optionally clear the one-shot one, then do the injection. Everything but doing the injection is implemented in the enter() method, it's perfectly worth re-using one.

### CODE REVIEW

File: utils/error_injection.hh [hunk 1]
```diff
@@ -431,15 +416,11 @@ public:
     requires std::is_invocable_r_v<std::exception_ptr, Func>
     [[gnu::always_inline]]
     future<>
-    inject(const std::string_view& name,
-            Func&& exception_factory) {
-
-        if (!is_enabled(name)) {
+    inject(const std::string_view& name, Func&& exception_factory) {
+        if (!enter()) {
```
Comment (tchaikov) [inline utils/error_injection.hh:420]: `name` is not passed to `enter()`.

Comment (xemul) [reply utils/error_injection.hh:420]: :facepalm:

**Comment by xemul** [comment]
upd:
- fixed lost `enter()` invocation w/o name argument

**Review by tchaikov — APPROVED** [review]
lgtm

**Pull request closed without merging**